experimental/Origami/src/canonical.jl

```diff
@@ -0,0 +1,69 @@
+# HACK HACK HACK special case optimization...
+function GAP.GapObj(obj::Vector{Int}; recursive::Bool = false)
+    len = length(obj)
+    ret_val = GAP.NewPlist(len)
+    for i = 1:len
+        ret_val[i] = obj[i]
+    end
+    return ret_val
+end
+
+function normal_form(o::Origami)
+    n = degree(o)
+    sym = symmetric_group(n)
+
+    x = sym(horizontal_perm(o))
+    y = sym(vertical_perm(o))
```

Comment:
```suggestion
    x = sym(horizontal_perm(o))
    y = sym(vertical_perm(o))
    sym = parent(x)
```

(I guess instead of writing `parent(x)` it would be nicer if there was something like `perm_group(o)`?